File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
vector<int> twoSum(vector<int>& numbers, int target)
{
    int idx1  =0 , idx2 = numbers.size()-1 ;
    while(idx1 <numbers.size() && idx2 >=0 )
    {
        if( numbers[idx1] +numbers[idx2] ==target )
            break;
        else if( numbers[idx1] +numbers[idx2] > target )
            idx2--;
        else if( numbers[idx1] +numbers[idx2] <target )
            idx1++;
    }
    return vector<int>  {idx1+1,idx2+1} ;
}
};

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.js
/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (numbers, target) {
  let idx1 = 0,
    idx2 = numbers.length - 1;
  while (idx1 < idx2) {
    if( numbers[idx1] + numbers[idx2 ] === target )
    break;
    else if(numbers[idx1] + numbers[idx2 ] > target )
    idx2--;
    else 
    idx1++;
  }
  return [idx1+1,idx2+1];
};
